src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -467,6 +507,12 @@ internal void Log(RpcEvent msg)
             }
         }
 
+        internal void HandleManagedDependencyError(Exception exc)
+        {
+            _workerChannelLogger.LogError(exc, $"{_workerConfig.Language} language worker failed to process dependency download request.");
+            _eventManager.Publish(new WorkerErrorEvent(_workerConfig.Language, Id, exc));
```

Comment:
what is expected behavior if the download fails by the language worker?  Should this surfaced to user as part of function load errors? If yes, defining a new exception makes more sense

Reply:
as powershell worker is avaialble to host, add E2E tests
- that succeed in downloading dependencies
- fail to download

Reply:
Did you miss this? Can you explain what is the expected behavior if dependency download fails?